src/main.rs

```diff
@@ -135,46 +132,36 @@ fn enclosure_overview(option: &ArgMatches) -> Result<(), ()> {
     Ok(())
 }
 
-/// TODO: Rework error handling, perhaps we don't need return Result 
+/// This function forks another binary for the prometheus-exporter.
 ///
-/// Returns an empty Result for now.
+/// # Arguments
 ///
-/// This function forks another binary for the prometheus-exporter. 
+/// * `option` - clappy's [ArgMatches]
 ///
-/// # Arguments
+/// # Errors
 ///
-/// * `option` - clappy's ArgMatches
+/// Errors on fork or waitpid are really rare it's bubble up as a boxed error.
 ///
-fn fork_prometheus(option: &ArgMatches) -> Result<(), ()> {
-    let mut default_port = "9945";
-    let mut default_address = "0.0.0.0";
-
-    if let Some(port) = option.value_of("port") {
-        default_port = port;
-    }
-
-    if let Some(ip) = option.value_of("ip-address") {
-        default_address = ip;
-    }
+fn fork_prometheus(option: &ArgMatches) -> Result<(), Box<dyn std::error::Error>> {
+    let default_port = option.value_of("port").unwrap_or("9945");
+    let default_address = option.value_of("ip-address").unwrap_or("0.0.0.0");
 
+    // SAFETY: fork is... unsafe by nature see man fork() thread issues etc.
     match unsafe { fork() } {
         Ok(ForkResult::Parent { child }) => {
             println!("prometheus-exporter pid: {:?}", child);
-            waitpid(Some(child), None).unwrap();
+            waitpid(Some(child), None)?;
             exit(0);
         }
 
         Ok(ForkResult::Child) => {
             Command::new(Util::JBOD_EXPORTER)
                 .args(&[default_address, default_port])
-                .spawn()
-                .expect("Failed to spawn the target process");
+                .spawn()?;
             exit(0);
         }
-        Err(_) => println!("Fork Failed"),
+        Err(e) => Err(Box::new(e)),
```

Comment:
Boxing error allows merging error together if the Error implements `std::error::Error` would not do that in a lib but in cli-app it's the way to go. A more powerful way in an app is anyhow like said above.

https://doc.rust-lang.org/rust-by-example/error/multiple_error_types/boxing_errors.html